### DISCUSSION TITLE
NullReferenceException: Object reference not set to an instance of an object

### DISCUSSION THREAD

**Discussion by misso2020** [aesim-tech/simba-project#197] [General]
Hi, The issue I'm talking about happened :
1- When importing simba license  by the command : 
from aesim.simba import License
License.Activate(''*****************************'')

I got 1 as a result  instead of 3 
2- When I Tried to execute the following flyback converter  : 
# Load modules
from aesim.simba import DesignExamples
# Load project
flybackConverter = DesignExamples.DCDC_Flyback()
# Get the job object and solve the system
job = flybackConverter.TransientAnalysis.NewJob()
status = job.Run()
# Get results
t = job.TimePoints
Vout = job.GetSignalByName('R2 - Instantaneous Voltage').DataPoints

I got the following error:

---------------------------------------------------------------------------
NullReferenceException                    Traceback (most recent call last)
<ipython-input-2-539277ca2109> in <module>
      3 
      4 # Load project
----> 5 flybackConverter = DesignExamples.DCDC_Flyback()
      6 
      7 # Get the job object and solve the system

NullReferenceException: Object reference not set to an instance of an object
  at Simba.Data.DesignExamples.DCDC_Flyback () [0x000c5] in <6d6c7685de16470ea4ea0cb1898cb605>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0003b] in <7cd4deb8302a478d9d393f24754841da>:0 

I tried several attempts to solve this issue  but nothing works for me. I use anaconda python 7 , and Kubuntu linux as operating  system
Any ideas to solve this problem?

![image](https://user-images.githubusercontent.com/61501735/153647238-a0fb34f4-c465-4eaa-9892-731f361c517e.png)

**Comment by aesim-tech** [comment]
Hello,

At the `License.Activate` Line, you need to include your Deployment Key (you will find it on your profile page). 

If you Deployment key is 123, you have to enter: 
`License.Activate("123")
`

I just tried on my side and I am not able to reproduce your error using a valid deployment key.  I will close this ticket for now but feel free to reopen it if the problem is not solved on your side.

**Comment by misso2020** [comment]
Thanks, but i covered my deployment key in purpose, its not a license problem its problem of having null values somewhere in the script , I don't know how to solve it.

**Comment by aesim-tech** [comment]
Can you please send your ´License.Activate(…)´ line code to ***@***.***?
I will try it here and see what’s going on. Other than that, I am running
the exact same code on Linux without any issues.

Also, which version of aesim.simba are you using? Make sure to use the
latest.

Finally, this issue can be causes by an internet connection restriction on
the machine. Does it have internet access?

**Comment by misso2020** [comment]
Thanks again, 
I sent my license to your email.
My aesim.simba package is 2021.12.2 and I have full access to the internet, I tried several times to make it work but no way, the interesting thing that even if I put anything in Activate.License(' ') , it outputs the same error.

![image](https://user-images.githubusercontent.com/61501735/153705976-41b4efe1-b7f1-4ff0-86cd-4061cada2bd9.png)

Also when  I comment from aesim.simba import DesignExamples, I got the same message.

![image](https://user-images.githubusercontent.com/61501735/153706070-bbce4ddf-c57f-44b7-8704-89fea18aa0ca.png)

Any suggestions please.

**Comment by aesim-tech** [comment]
Hello,

I sent you an email last week. Have you received it? Is your problem solved? Unfortunately, I am not able to reproduce the bug.

**Comment by misso2020** [comment]
Thanks so much Emmanuel , I received your email and I answered it, till now the issue couldn't be solved, I  uninstalled anaconda 3 entirely  with all its components, then I reinstalled it but the problem is always there , I made a lot of search in the net , but no way. I hope that I can find some help in the near future.
Thanks

**Comment by aesim-tech** [comment]
Messaoud,

Which Linux Distribution are you using? Is it a physical or virtual
machine? If it is virtual, where is it hosted (AWS... Google...)?
If you want to troubleshoot that issue with me, can you please Book a
meeting with me <https://meetings.hubspot.com/manu27>?

Thanks,
Emmanuel.

On Sun, Feb 20, 2022 at 2:27 PM Messaoud Makhlouf ***@***.***>
wrote:

> Thanks so much Emmanuel , I received your email and I answered it, till
> now the issue couldn't be solved, I uninstalled anaconda 3 entirely with
> all its components, then I reinstall it but the problem is always there , I
> made a lot of search in the net , but no way. I hope that I can find some
> help in the near future.
> Thanks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aesim-tech/simba-project/issues/176#issuecomment-1046304701>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APAGJS5YO4WTZY4K3XKCBUTU4E6A3ANCNFSM5OFAMDIA>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>


-- 
Emmanuel RUTOVIC
(438) 926-6458

**Comment by misso2020** [comment]
Dear Emmanuel,

Thanks a lot for your time and effort for solving issues related to simba software, I'm glad to inform you that this evening I solved my problem and started using aesim.simba package on  kubuntu 21.10 without problem.
Many thanks for your proposition of booking a meeting.

**Comment by aesim-tech** [comment]
Hello Messaoud,

Thanks for sharing this good news! I'll close the ticket for now. 

Regards,
Emmanuel.
